pymc3_hmm/utils.py

```diff
@@ -102,10 +104,12 @@ def tt_logdotexp(A, b):
     The result is more or less equivalent to `tt.log(tt.exp(A).dot(tt.exp(b)))`
 
     """
-    A_bcast = A.dimshuffle(list(range(A.ndim)) + ["x"])
+    pattern: List[Any] = list(range(A.ndim))
+    A_bcast = A.dimshuffle(pattern + ["x"])
 
     sqz = False
-    shape_b = ["x"] + list(range(b.ndim))
+    pattern_b: List[Any] = list(range(b.ndim))
+    shape_b: List[Any] = ["x"] + pattern_b
     if len(shape_b) < 3:
```

Comment:
I feel like there's a more elegant solution than this "pattern" hack but I was having trouble finding another workaround for the original errors:

`Argument 1 to "list" has incompatible type "range"; expected "Iterable[str]"` (line 108)
`List item 0 has incompatible type "str"; expected "int"` (line 105)

Reply:
It looks like `shape_b` should be typed as `shape_b: List[Union[Text, int]]`.

Reply:
gave `List[Union...` a shot but it raised a new error `error: Incompatible types in assignment (expression has type "List[str]", variable has type "List[Union[str, int]]")`

Reply:
Ah, yeah, `List[Text]` isn't a subtype of `List[Union[Text, int]]` due to [invariance](https://mypy.readthedocs.io/en/latest/common_issues.html#invariance-vs-covariance).

Reply:
ok I went back to what I originally had here since it doesn't raise any mypy errors, unless there are any other suggestions on how to handle this better?

Reply:
As always, we want to avoid using `Any`, especially when we know what the types are (and they're not literally anything), because that tends to defeat the purpose/usefulness static typing in these situations.

Regarding this invariance issue, we might be able to change the `List` to a `Sequence`, which I believe is covariant.

Reply:
Is the idea with `Sequence` to try something like this?

```python
shape_b : Sequence[Union[Text, int]] = ["x"] + list(range(b.ndim))
```

When I tried that it threw this error:

```python
pymc3_hmm/utils.py:115: error: Argument 1 to "list" has incompatible type "range"; expected "Iterable[str]"  [arg-type]
pymc3_hmm/utils.py:115: note: Following member(s) of "range" have conflicts:
pymc3_hmm/utils.py:115: note:     Expected:
pymc3_hmm/utils.py:115: note:         def __iter__(self) -> Iterator[str]
pymc3_hmm/utils.py:115: note:     Got:
pymc3_hmm/utils.py:115: note:         def __iter__(self) -> Iterator[int]
```

Reply:
It seems like the solution to this could get complicated, so let's go with the `Any` approach for now.